modules/web/src/app/shared/components/application-list/component.ts

```diff
@@ -278,64 +284,91 @@ export class ApplicationListComponent implements OnInit, OnDestroy {
   }
 
   private _updateApplicationMaps(): void {
-    this.applicationsMethodMap = {};
-    this.applicationsSourceMap = {};
-    this.applicationsStatusMap = {};
-    this.applications.forEach(application => {
-      const applicationRef = application.spec.applicationRef;
-      const status = application.status;
-      const namespace = application.spec.namespace.name;
-
-      this.applicationsMethodMap = {
-        ...this.applicationsMethodMap,
-        [namespace]: {
-          ...(this.applicationsMethodMap[namespace] || {}),
-          [application.name]: status?.method || this.applicationDefinitionsMap.get(applicationRef?.name)?.spec.method,
-        },
-      };
-
-      const versionSources =
-        status?.applicationVersion?.template?.source ||
-        this.applicationDefinitionsMap
-          .get(applicationRef?.name)
-          ?.spec.versions?.find(version => version.version === applicationRef?.version)?.template?.source;
-
-      this.applicationsSourceMap = {
-        ...this.applicationsSourceMap,
-        [namespace]: {
-          ...(this.applicationsSourceMap[namespace] || {}),
-          [application.name]: versionSources ? Object.keys(versionSources).find(key => !!versionSources[key]) : '',
-        },
-      };
-      if (status) {
-        let icon = StatusIcon.Pending;
-        let message = '';
-        if (application.deletionTimestamp) {
-          icon = StatusIcon.Error;
-          message = 'Deleting';
+    const loadingAppDefDetails: Record<string, boolean> = {};
+    // load application definition details
+    forkJoin(
+      this.applications.map(application => {
```

Comment:
Just curious why do we need this `loadingAppDefDetails` local  Record i guess on every onChanges this will be new field on memory and storing reference to check if application definition is loaded or not doesn't serve purpose IMO or is it? 

Secondly, instead of complex conditional checking to load definition whyn't simply load definition for each application in application list to simplify code?

Reply:
This method is called from `ngOnChanges` when `applications` list is updated so that means some application is added/updated/removed.
Secondly, there can be multiple applications point to same application definition so I added `loadingAppDefDetails` to avoid duplicate API calls (i.e. to avoid loading details of same application definition multiple times).

Also, we want to load details of only those application definitions which haven't been loaded yet.